src/lock_file/conda_prefix_updater.rs

```diff
@@ -0,0 +1,125 @@
+use super::utils::IoConcurrencyLimit;
+use super::PixiRecordsByName;
+use crate::build::BuildContext;
+use crate::environment::{self, PythonStatus};
+use crate::prefix::Prefix;
+use crate::project::grouped_environment::{GroupedEnvironment, GroupedEnvironmentName};
+use crate::project::HasProjectRef;
+use futures::TryFutureExt;
+use miette::IntoDiagnostic;
+use pixi_manifest::FeaturesExt;
+use rattler::package_cache::PackageCache;
+use rattler_conda_types::Platform;
+use std::sync::Arc;
+use tokio::sync::Semaphore;
+
+/// A struct that contains the result of updating a conda prefix.
+pub struct CondaPrefixUpdated {
+    /// The name of the group that was updated.
+    pub group: GroupedEnvironmentName,
+    /// The prefix that was updated.
+    pub prefix: Prefix,
+    /// Any change to the python interpreter.
+    pub python_status: Box<PythonStatus>,
+}
+
+#[derive(Clone)]
+/// A task that updates the prefix for a given environment.
+pub struct CondaPrefixUpdater<'a> {
+    pub group: GroupedEnvironment<'a>,
+    pub package_cache: PackageCache,
+    pub io_concurrency_limit: IoConcurrencyLimit,
+    pub build_context: BuildContext,
+    pub no_install: bool,
+}
+
+impl<'a> CondaPrefixUpdater<'a> {
+    /// Creates a new prefix task.
+    pub fn new(
+        group: GroupedEnvironment<'a>,
+        package_cache: PackageCache,
+        io_concurrency_limit: IoConcurrencyLimit,
+        build_context: BuildContext,
+        no_install: bool,
+    ) -> Self {
+        Self {
+            group,
+            package_cache,
+            io_concurrency_limit,
+            build_context,
+            no_install,
+        }
+    }
+
+    /// Updates the prefix for the given environment.
+    pub(crate) async fn update(
+        &self,
+        records: Arc<PixiRecordsByName>,
+    ) -> miette::Result<CondaPrefixUpdated> {
```

Comment:
Please make an error type

Reply:
Sorry, will not be able to this in this PR as it would result in needing to refactor a lot of Error types in update. We need to this seperately. I can take the message from miette  for `update_prefix_conda` and `find_installed_packages` and put them in an error but I do not believe this to be advantageous at this point.